grid2op/tests/test_RewardAlertCostScore.py

```diff
@@ -0,0 +1,201 @@
+        return super().act(observation, reward, done)
+
+
+class TestAlertCostScore(unittest.TestCase):
+    
+    def test_specs(self):
+        # test function without actual data
+        assert _AlertCostScore._penalization_fun(50) == -1.
+        assert _AlertCostScore._penalization_fun(80) == 0.
+        assert _AlertCostScore._penalization_fun(100) == 1.
+    
+    def setUp(self) -> None:
+        self.env_nm = os.path.join(
+            PATH_DATA_TEST, "l2rpn_idf_2023_with_alert"
+        )
+        
+    def tearDown(self) -> None:
+        return super().tearDown()
+        
+    def test_assistant_reward_value_no_blackout_no_attack_no_alert(self) -> None : 
+        """ When no blackout and no attack occur, and no alert is raised we expect a reward of 0
+            until the end of the episode where we get the max reward 1.
+
+        Raises:
+            Grid2OpException: raise an exception if an attack occur
+        """
+        with grid2op.make(
+            self.env_nm,
+            test=True,
+            difficulty="1",
+            reward_class=_AlertCostScore
+        ) as env:
+            env.seed(0)
+            env.reset()
+
+            done = False
+            for i in range(env.max_episode_duration()):
+                obs, reward, done, info = env.step(env.action_space())
+                if done:
+                    assert reward == 1.
+                else:
+                    assert reward == 0.
+                    
+                    
+class TestSimulate(unittest.TestCase):
+    def setUp(self) -> None:
+        self.env_nm = os.path.join(
+            PATH_DATA_TEST, "l2rpn_idf_2023_with_alert"
+        )
+        self.env = grid2op.make(self.env_nm, test=True, difficulty="1",
+                        reward_class=_AlertCostScore)
+        self.env.seed(0)
+        return super().setUp()
+    
+    def tearDown(self) -> None:
+        self.env.close()
+        return super().tearDown()
+    
+    def test_simulate(self):
+        obs = self.env.reset()
+        simO, simr, simd, simi = obs.simulate(self.env.action_space())
+        assert simr == 0.
+        assert not simd
+        
+        go_act = self.env.action_space({"set_bus": {"generators_id": [(0, -1)]}})
+        simO, simr, simd, simi = obs.simulate(go_act)
+        assert simr == 0., f"{simr} vs 0."
+        assert simd
+        
+    def test_simulated_env(self):
+        obs = self.env.reset()
+        f_env = obs.get_forecast_env()
+        forD = False
+        while not forD:
+            forO, forR, forD, forI = f_env.step(self.env.action_space())
+            assert forR == 0.
+            
+        f_env = obs.get_forecast_env()
+        forD = False
+        go_act = self.env.action_space({"set_bus": {"generators_id": [(0, -1)]}})
+        while not forD:
+            forO, forR, forD, forI = f_env.step(go_act)
+            assert forR == 0.
+    
+    
+class TestRunner(unittest.TestCase):
```

Comment:
Can you rename it "TestRunnerAlertCost" ? And do a "TestRunnerAlertTrust" too ?

Reply:
ok !